bookkeeper-http/servlet-http-server/src/main/java/org/apache/bookkeeper/http/servlet/BookieHttpServiceServlet.java

```diff
@@ -0,0 +1,117 @@
+
+  public BookieHttpServiceServlet(){
+    HttpRouter<ApiType> router = new HttpRouter<HttpServer.ApiType>(
+      new AbstractHttpHandlerFactory<ApiType>(BookieServletHttpServer.getBookieHttpServiceProvider()) {
+        @Override
+        public HttpServer.ApiType newHandler(HttpServer.ApiType apiType) {
+          return apiType;
+        }
+      }) {
+      @Override
+      public void bindHandler(String endpoint, HttpServer.ApiType mapping) {
+        mappings.put(endpoint, mapping);
+      }
+    };
+    router.bindAll();
+  }
+
+  @Override
+  protected void service(HttpServletRequest httpRequest, HttpServletResponse httpResponse) throws ServletException, IOException {
+    HttpServiceRequest request = new HttpServiceRequest()
+                                .setMethod(httpServerMethod(httpRequest))
+                                .setParams(httpServletParams(httpRequest))
+                                .setBody(IOUtils.toString(httpRequest.getInputStream(), "UTF-8"));
+    String uri = httpRequest.getRequestURI();
+    HttpServiceResponse response;
+    try {
+      HttpServer.ApiType apiType = mappings.get(uri);
+      HttpServiceProvider bookie = BookieServletHttpServer.getBookieHttpServiceProvider();
+      if (bookie == null) {
+        httpResponse.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        return;
+      }
+      HttpEndpointService httpEndpointService = bookie.provideHttpEndpointService(apiType);
+      if (httpEndpointService == null) {
+        httpResponse.sendError(HttpServletResponse.SC_NOT_FOUND);
+        return;
+      }
+      response = httpEndpointService.handle(request);
+    } catch (Throwable e) {
+      LOG.error( "Error while service Bookie API request " + uri, e);
+      response = new ErrorHttpService().handle(request);
+    }
+    if (response != null) {
+      httpResponse.setStatus(response.getStatusCode());
+      try (Writer out = httpResponse.getWriter()) {
+        out.write(response.getBody());
+      }
+    } else {
+      httpResponse.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    }
+  }
+
+
+  /**
+   * Convert http request parameters to a map.
+   */
+  @SuppressWarnings("unchecked")
+  Map<String, String> httpServletParams(HttpServletRequest request) {
+    Map<String, String> map = new HashMap<>();
+    for (Enumeration<String> param = request.getParameterNames();
+         param.hasMoreElements();) {
+      String pName = param.nextElement();
+      map.put(pName, request.getParameter(pName));
+    }
+    return map;
+  }
+
+  /**
+   * Get servlet request method and convert to the method that can be recognized by HttpServer.
+   */
+  HttpServer.Method httpServerMethod(HttpServletRequest request) {
+    switch (request.getMethod()) {
+      case "POST":
+        return HttpServer.Method.POST;
+      case "DELETE":
+        return HttpServer.Method.DELETE;
+      case "PUT":
+        return HttpServer.Method.PUT;
+      default:
```

Comment:
Please implement explicitly GET and throw for other cases

Reply:
> Please implement explicitly GET and throw for other cases

done